mogp_emulator/MultiOutputGP.py

```diff
@@ -101,6 +106,81 @@ def __init__(self, inputs, targets, mean=None, kernel="SquaredExponential", prio
                            for (single_target, m, k, p, n) in zip(targets, mean, kernel, priorslist, nugget)]
 
 
+    @property
+    def inputs(self):
+        """Full array of emulator inputs
+        
+        Returns input array (2D array of shape
+        ``(n, D)``).
+        
+        :returns: Array of input values
+        :rtype: ndarray
+        """
+        return self.emulators[0].inputs
+
+    @property
+    def targets(self):
+        """Full array of emulator targets
+        
+        Returns target array (2D array of shape
+        ``(n_emulators, n)``).
+        
+        :returns: Array of target values
+        :rtype: ndarray
+        """
+        targetlist = []
+        for em in self.emulators:
+            targetlist.append(em.targets)
+        return np.array(targetlist)
+
+    @property
+    def D(self):
+        """Number of Dimensions in inputs
+        
+        Returns number of dimentions of the input data
+        
+        :returns: Number of dimentions of inputs
+        :rtype: int
+        """
+        return self._D
+
+    @property
+    def n(self):
+        """Number of training examples in inputs
+        
+        Returns length of training data.
+        
+        :returns: Length of training inputs
+        :rtype: int
+        """
+        return self._n
+
+    @property
+    def n_params(self):
+        """Returns the number of parameters for all emulators
+        as a list of integers
+        
+        :returns: Number of parameters for each emulator
+                  as a list of integers
+        :rtype: list
+        """
+        return [em.n_params for em in self.emulators]
```

Comment:
At the moment, the GPU implementation assumes that all emulators will have the same number of parameters, and just returns an int - is this something I should change in the GPU implementation?

Reply:
Sounds good, I made and pushed the corresponding change for GPU now, so everything should match up.